hc-tree-node.html

```diff
@@ -0,0 +1,129 @@
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../iron-icon/iron-icon.html">
+<link rel="import" href="../iron-icons/iron-icons.html">
+<link rel="import" href="../paper-tristate-checkbox/paper-tristate-checkbox.html">
+<link rel="import" href="hc-nested-list-item.html">
+
+<!--
+`hc-tree-node`
+A node for the tree view, handles the basic node template with proper tristate selection handling.
+
+@demo demo/index.html
+-->
+<dom-module id="hc-tree-node">
+  <template>
+    <style>
+      :host {
+        display: block;
+      }
+    </style>
+
+    <hc-nested-list-item item="{{item}}">
+      <slot name="checkbox">
+        <paper-tristate-checkbox
+          id="checkbox"
+          state="{{item.state}}"
+          value="{{_value}}"
+          checked="[[checked]]"
+          unchecked="[[unchecked]]"
+          indeterminate="[[indeterminate]]"
+          on-state-changed="_stateChanged"
+        ></paper-tristate-checkbox>
+      </slot>
+      <slot name="icon">
+        <iron-icon icon="icons:add-circle-outline"></iron-icon>
+      </slot>
+      <slot name="label">
+        <span>[[item.label]]</span>
+      </slot>
+    </hc-nested-list-item>
+  </template>
+
+  <script>
+    Polymer({
+      /* global hc */
+      is: 'hc-tree-node',
+      properties: {
+        /* Used to change selected state when checkbox is checked */
+        checked: {
+          type: Boolean,
+          value: true
+        },
+        /* Used to change selected state when checkbox is indterminate */
+        indeterminate: {
+          type: Boolean,
+          value: false
+        },
+        /* This node's data */
+        item: {
+          type: Object,
+          notify: true
+        },
+        /* Used to change selected state when checkbox is unchecked */
+        unchecked: {
+          type: Boolean,
+          value: false
+        },
+        /* Holds the value that determins selected status for the item */
+        _value: {
+          type: Boolean,
+          observer: '_valueChanged'
+        },
+        __init: {
+          type: Boolean,
+          value: true
+        }
+      },
+      ready() {
+        /**
+         * Overload the default tap functionality to prevent user clicks from setting indeterminate state,
+         *  and to prevent event bubbling.  This is using an annonymous function to keep the correct scope
+         *  that is provided by the checkbox element instead of an arrow function which will use the local
+         *  scope.
+         */
+        this.$.checkbox._regularTap = function(event) {
+          // event.stopPropagation();
+          this.state = this.state === 'on' ? 'off' : 'on';
+        };
+      },
+      /**
+       * Watches for changes to the value for the checkbox then notifies the parent of selection changes.
+       *
+       * @fires hc-tree-view-behavior#item-state-changed
+       * @param {CustomEvent} event
+       * @param {Object} detail
```

Comment:
No benefit, just something that's provided for convenience, and it's the only part of event that I actually care about.